src/hdf5_time_series_storage.jl

```diff
@@ -115,6 +115,30 @@ function serialize_time_series!(
     return
 end
 
+function get_data_type(ts::TimeSeriesData)
```

Comment:
I suggest calling `eltype` only once.  Also, you shouldn't use `.data`. This will not work in my current PR.  Instead, I think we need a function implemented on the concrete types.
```
eltype_data(ts)
```